muttlib.c

```diff
@@ -488,26 +488,30 @@ void mutt_mktemp_full(char *s, size_t slen, const char *prefix,
                strerror(errno), errno);
 }
 
-FILE *mutt_mkstemp_full(char *s, size_t slen, const char *src, int line)
+FILE *mutt_mkstemp_full(const char *src, int line)
 {
+  char s[_POSIX_PATH_MAX];
   int fd;
   FILE *fp;
 
-  size_t n = snprintf(s, slen, "%s/neomutt-XXXXXX", NONULL(Tmpdir));
+  int n = snprintf(s, sizeof(s), "%s/neomutt-XXXXXX", NONULL(Tmpdir));
 
   if (n < 0)
-    mutt_debug(1, "%s:%d: snprintf failed! returns \"%s\".\n", src, line, n);
+    mutt_debug_real(__func__, __FILE__, __LINE__, 1,
+                    "%s:%d: snprintf failed! returns \"%s\".\n", src, line, n);
```

Comment:
Hmm...  Sorry, I've made things worse.
How about:
- After the `snprintf()` we `mutt_debug_real()` with the name of the file.
  This means the log shows the caller and the filename.
- The rest of the function can use `mutt_debug()` as before.
  If anything goes wrong, we see the line in `mutt_mkstemp_full()`